src/Compilers/Core/Portable/SourceGeneration/IncrementalGeneratorInitializationContext_ForAttributeWithMetadataName.cs

```diff
@@ -0,0 +1,154 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Immutable;
+using System.Linq;
+using System.Xml.Serialization;
+using Microsoft.CodeAnalysis.PooledObjects;
+using Roslyn.Utilities;
+using System.Collections;
+using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
+using Microsoft.CodeAnalysis.SourceGeneration;
+using System.Threading;
+
+namespace Microsoft.CodeAnalysis;
+
+using Aliases = ArrayBuilder<(string aliasName, string symbolName)>;
+
+internal readonly struct GeneratorAttributeSyntaxContext
+{
+    /// <summary>
+    /// The syntax node the attribute is attached to.  For example, with <c>[CLSCompliant] class C { }</c> this would
+    /// the class declaration node.
+    /// </summary>
+    public SyntaxNode AttributeTarget { get; }
+
+    /// <summary>
+    /// Semantic model for the file that <see cref="AttributeTarget"/> is contained within.
+    /// </summary>
+    public SemanticModel SemanticModel { get; }
+
+    /// <summary>
+    /// <see cref="AttributeData"/>s for any matching attributes on <see cref="AttributeTarget"/>.  Always non-empty.
+    /// </summary>
+    public ImmutableArray<AttributeData> Attributes { get; }
```

Comment:
note: this changed slightly now that we pass AttributeData along.  A node might have multiple attributes on it with the same matching metadata name.  so you are given all the matching attributes with the single target at the same time.  an alternative would be to get multiple callbacks for the same target with each attribute, but this seemed fine to me.

Reply:
i've resolved this by exposing the ISymbol the attributes belong to.  This way you can see both views. just the matching attributes, and all the attributes.